### PR TITLE
[travis] add flake8 to travis

### PR DESCRIPTION
**Pull request by jnozsc**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [x] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/ytdl-org/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/ytdl-org/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/ytdl-org/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [x] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Explanation of your *pull request* in arbitrary form goes here. Please make sure the description explains the purpose and effect of your *pull request* and is worded well enough to be understood. Provide as much context and examples as possible.

enforce `flake8` test in travis

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -13,7 +13,7 @@ dist: trusty
 env:
   - YTDL_TEST_SET=core
   - YTDL_TEST_SET=download
-matrix:
+jobs:
```
Comment (dstftw) [inline .travis.yml]: What's the point of this?

File: .travis.yml [hunk 2]
```diff
@@ -42,4 +42,7 @@ matrix:
     - env: JYTHON=true; YTDL_TEST_SET=download
 before_install:
   - if [ "$JYTHON" == "true" ]; then ./devscripts/install_jython.sh; export PATH="$HOME/jython/bin:$PATH"; fi
-script: ./devscripts/run_tests.sh
+script:
+  - pip install "flake8; python_version >= '2.7' and python_version != '3.0' and python_version != '3.1' and python_version != '3.2' and  python_version != '3.3'"
+  - if [ -x "$(command -v flake8)" ]; then flake8 . ;fi
```
Comment (dstftw) [inline .travis.yml]: This must be executed one single time in a separate job. Running this in each job is just useless.

File: .travis.yml (hunk 1 above)
Comment (jnozsc) [reply .travis.yml]: `jobs` is the new key, `matrix` becomes the alias of it

https://github.com/travis-ci/travis-yml/blob/master/schema.json#L546

File: .travis.yml (hunk 2 above)
Comment (jnozsc) [reply .travis.yml]: fixed

**Pull request merged**